deepchem/data/datasets.py

```diff
@@ -1092,7 +1274,13 @@ def get_task_names(self) -> np.ndarray:
     return self.tasks
 
   def reshard(self, shard_size: int) -> None:
-    """Reshards data to have specified shard size."""
+    """Reshards data to have specified shard size.
+
+    Note
+    ----
+    If this `DiskDataset` is in `legacy_metadata` format, reshard will
+    maintain legacy metadata format.
```

Comment:
Is there any reason not to upgrade it to the new format?  That will give better performance.

Reply:
Good point. I'll upgrade it to the new format